src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -305,6 +307,13 @@ private int WriteTags(ref byte[] bytes, int offset, Span span)
                 }
             }
 
+            if (span.IsTopLevel)
+            {
+                count++;
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _runtimeIdBytes);
```

Comment:
I like this idea of adding tags during serialization instead of adding them to the in-memory object. We should adopt this for other cases, like global tags added to every span (`DD_TAGS`).